WCF/Shared/RequestTrackingTelemetryModule.cs

```diff
@@ -85,7 +85,12 @@ void IWcfTelemetryModule.OnEndRequest(IOperationContext operation, Message reply
                 responseCode = HttpStatusCode.InternalServerError;
             }
 
-            telemetry.Success = !isFault;
+            // if the operation code has already marked the request as failed
+            // don't overwrite the value if we think it was successful
+            if ( isFault || !telemetry.Success.HasValue )
+            {
+                telemetry.Success = !isFault;
+            }
             telemetry.ResponseCode = responseCode.ToString("d");
```

Comment:
I was thinking for consistency. Also some people may want to override it with non-HTTP result codes (something more application specific)